File: app/src/main/java/com/szeptun/shoppinglist/domain/SaveShoppingList.java
package com.szeptun.shoppinglist.domain;

import com.szeptun.shoppinglist.dataaccess.repository.ShoppingListRepository;
import com.szeptun.shoppinglist.entity.ShoppingList;

import javax.inject.Inject;

import io.reactivex.Completable;

public class SaveShoppingList {

    @Inject
    SaveShoppingList() {
    }

    @Inject
    ShoppingListRepository repository;

    public Completable execute(ShoppingList shoppingList) {
        return repository.saveList(shoppingList);
    }
}

File: app/src/main/java/com/szeptun/shoppinglist/domain/SaveShoppingList.kt
package com.szeptun.shoppinglist.domain

import com.szeptun.shoppinglist.dataaccess.repository.ShoppingListRepository
import com.szeptun.shoppinglist.entity.ShoppingList
import io.reactivex.Completable
import javax.inject.Inject

class SaveShoppingList @Inject constructor(private val repository: ShoppingListRepository) {

    fun execute(shoppingList: ShoppingList): Completable =
        repository.saveList(shoppingList)
}